source/index.ts

```diff
@@ -39,21 +37,67 @@ const checkValueType = (key, value) => {
 const INTERNAL_KEY = '__internal__';
 const MIGRATION_KEY = `${INTERNAL_KEY}.migrations.version`;
 
-class Conf {
-	constructor(options) {
+type ConfSerializer = (args: any) => ArrayBuffer | Buffer | string;
+type ConfDeserializer = (args: any) => any;
+type ConfDefaultValues = {
+	[key: string]: object;
+};
+type ConfMigrations = {
+	[key: string]: (store: Conf) => void;
+};
+type GenericCallback = () => any;
+type GenericVoidCallback = (...args: any) => void;
+type StoreValue = any;
+
+type ConfOptions = {
+	accessPropertiesByDotNotation?: boolean;
+	clearInvalidConfig?: boolean;
+	configName?: string;
+	cwd?: string;
+	defaults?: object;
+	encryptionKey?: string;
+	fileExtension?: string;
+	migrations?: ConfMigrations;
+	projectName?: string;
+	projectSuffix?: string;
+	projectVersion?: string;
+	schema?: object;
+	watch?: boolean;
+	serialize?: ConfSerializer;
+	deserialize?: ConfDeserializer;
+};
+
+export default class Conf {
+	_options: ConfOptions;
+
+	_defaultValues: ConfDefaultValues = {};
+
+	_validator?: Ajv.ValidateFunction;
+
+	encryptionKey?: string;
+
+	events?: EventEmitter;
```

Comment:
Why is this optional? Same with some of the others here.

Reply:
The only one was actually `_validator` 😅 

Nice catch